src/foam/nanos/auth/CapabilityAuthService.js

```diff
@@ -163,81 +163,36 @@ foam.CLASS({
               NOT(HAS(UserCapabilityJunction.EXPIRY)),
               NOT(EQ(UserCapabilityJunction.STATUS, CapabilityJunctionStatus.EXPIRED))
           );
-
-          if ( cap != null && cap.getEnabled() ) {
-
-            if ( userCapabilityJunctionDAO.find(
-              AND(
-                userPredicate,
-                capabilityScope,
-                EQ(UserCapabilityJunction.TARGET_ID, cap.getId()),
-                EQ(UserCapabilityJunction.STATUS, CapabilityJunctionStatus.GRANTED)
-              )) != null ) {
-              result = true;
-              // if the user has the permission, store this in the cache and return the result
-              // otherwise, move on to the 2nd part of the check
-              if ( result ) {
-                ((Map<String, Boolean>) getCache()).put(userKey, result);
-                return result;
-              }
-            } else if ( cap != null && agent != null && agent.getId() != user.getId() ) {
-              // if the agent is in the context and is not the same as user, check 
-              // if the agent has capability
-              userPredicate = AND(
-                INSTANCE_OF(AgentCapabilityJunction.class),
-                EQ(UserCapabilityJunction.SOURCE_ID, agent.getId()),
-                EQ(AgentCapabilityJunction.EFFECTIVE_USER, user.getId())
-              );
-              if ( userCapabilityJunctionDAO.find(
-                AND(
-                  userPredicate,
-                  capabilityScope,
-                  EQ(UserCapabilityJunction.TARGET_ID, cap.getId()),
-                  EQ(UserCapabilityJunction.STATUS, CapabilityJunctionStatus.GRANTED)
-                )) != null )
-                result = true;
-                // if the user has the permission, store this in the cache and return the result
-                // otherwise, move on to the 2nd part of the check
-                if ( result ) {
-                  if ( agentKey != null ) ((Map<String, Boolean>) getCache()).put(agentKey, result);
-                  return result;
-                }
-            }
-          }
-
-          // 2. check if the user has a capability that grants the permission
-          userPredicate = EQ(UserCapabilityJunction.SOURCE_ID, user.getId());
           AbstractPredicate predicate = new AbstractPredicate(x) {
             @Override
             public boolean f(Object obj) {
               UserCapabilityJunction ucj = (UserCapabilityJunction) obj;
               Capability c = (Capability) capabilityDAO.find(ucj.getTargetId());
-              if ( ucj.getStatus() == CapabilityJunctionStatus.GRANTED && 
+              if ( ucj.getStatus() == CapabilityJunctionStatus.GRANTED &&
                    c != null && ! c.isDeprecated(x) && c.implies(x, permission) ) {
                 return true;
               }
               return false;
             }
           };
+
+          // 1. check if the user has a capability that grants the permission
```

Comment:
I'd suggest either removing the numbers or consider making the top checks to the cache and the maybeIncercept #1. from lines 154-158

Reply:
Also would suggest a comment update:
//# Check if a ucj implies the user has this permission